hack/generator/spec/swagger-resource-references-onepager.md

```diff
@@ -0,0 +1,54 @@
+```
+    "SubResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "resourceId",
+          "description": "Resource ID."
+        }
+      },
+      "description": "Reference to another subresource."
+    },
+```
+
+## What is Azure Service Operator (ASO)?
+It's a Kubernetes operator which supports creating Azure resources. You can read more [here](https://github.com/Azure/azure-service-operator). It allows customers to create Azure resources directly in their Kubernetes cluster by defining YAML files representing the Azure resource and then doing `kubectl apply -f <my-yaml>`.
+
+## How do resource references work in ASO?
+ASO allows you to represent various Azure resources in Kubernetes. As a consequence of the fact that the resource is reflected in Kubernetes, you also may _refer_ to resources in two different ways:
+1. A reference to the resource via its ARM ID, the same as you would do in an ARM template or the SDK.
+2. A reference to the resource via its Kubernetes name. This is more natural to Kubernetes developers and plays to Kubernetes strengths by allowing you to more easily apply the same specification in multiple places (generating distinct resources with distinct ARM IDs) without having to update the resource specification for each kind of deployment. An example of this would be creating the same resource in Kubernetes in two different namesapces which map to two different subscriptions in Azure - the "Kubernetes reference" looks the same but the ARM ID of the actual referenced resource is different.
+
+You can read more about the references design [here](https://github.com/Azure/k8s-infra/blob/master/hack/generator/spec/type_references_and_ownership.md)
+
+## Why do we want this ability?
+In a lot of ways, the maintainers of ASO (that's my team and I) are in the same position as the SDK team: there are tons of Azure services and there's no way we can keep up with the pace of new API versions/services without some sort of automation helping us. We are currently most of the way through developing a code-generator that generates Kubernetes CRDs from the azure-rest-api-specs Swager documents - you can see this generator here: [k8sinfra](https://github.com/Azure/k8s-infra).
+
+In order for the generator to generate the right shape for resources that have references, we need a way to determine that a particular field is a reference, so that we can replace the simple `string` field with something like:
+```
+type ResourceReference struct {
+	// Group is the Kubernetes group of the resource.
+	Group string `json:"group"`
+	// Kind is the Kubernetes kind of the resource.
+	Kind string `json:"kind"`
+	// Namespace is the Kubernetes namespace of the resource.
+	Namespace string `json:"namespace"`
+	// Name is the Kubernetes name of the resource.
+	Name string `json:"name"`
+
+        // +kubebuilder:validation:Pattern="^/subscriptions/(.+?)/resourcegroups/(.+?)/providers/(.+?)/((.+?)/(.+?))+$"
```

Comment:
Not sure if this matters for the proposal but just FYI.  This isn't the format for all resourceIds.  It could be an id of a resource that lives under a subscription so the resourceGroups won't be there, it could be something that lives under a tenant, so subscription and resourcegroups won't be there.  it could be an extension resource where you have an entire resourceId in the middle of one of these resourceIds.  I believe this also assumes that the end consists of /ref/value/ pairs, but thats not always the case either you can have /ref1/ref2/value/ in fact the number of refs doesn't seem to have a limit.